egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
   this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
 
+  if [ "$adaptive_shrink" == "true" ]; then
+    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
```

Comment:
In addition to the rate of change, I am also worried about the initial shrink value. This, I believe, would affect learning to a greater extent. Further the shrink value is not going to reach 1 towards the end of the training in the above equation. We actually wanted $shrink to reach 1 as this would ensure we finally are optimizing the actual objective function.

So though we do not have an l2 regularization interpretation, how about keeping the original shrink value formula based on the effective_learning_rate rather than actual_learning_rate 
<code>
 this_shrink=$(perl -e "print $shrink + ($flr/($this_learning_rate/$this_num_jobs)) * (1 - $shrink)")
</code>